sdk/src/transaction_context.rs

```diff
@@ -204,34 +216,70 @@ impl TransactionContext {
     }
 
     /// Used by the runtime when a new CPI instruction begins
-    pub fn record_compiled_instruction(&mut self, instruction: CompiledInstruction) {
+    pub fn record_instruction(&mut self, stack_height: usize, instruction: InstructionContext) {
```

Comment:
Why have `stack_height` as parameter here?
`self.get_instruction_context_stack_height()` would also do it, right?

Reply:
The requires that at the time of recording that the active stack height be current  I think that is an unnecessary dependency enforced on the caller

Reply:
You mean the push might come earlier or later, ok then leave it as is.